### ISSUE TITLE
Date format is not working with management form output

### ISSUE THREAD

**Issue by breneser** [viewflow/viewflow#263]
Hi,

I am having a bit of an issue with management_form date format output. 

`{{activation.management_form}}` puts in a hidden input as `<input type="hidden" name="_viewflow_activation-started" value="2020-03-16 16:07:48.932914" id="id__viewflow_activation-started">`  

Based on date format in settings I was expecting `<input type="hidden" name="_viewflow_activation-started" value="16-03-2020 16:07:48" id="id__viewflow_activation-started">`. I am putting this hidden input in myself as a workaround as  management_form didn't output the correct date format.

When submitted I get Enter a valid date/time error. 

See date settings below:

```
# Date formats
DATE_INPUT_FORMATS = ['%d-%b-%Y']  # input formats
DATETIME_INPUT_FORMATS = ['%d-%b-%Y %H:%M:%S']
DATE_FORMAT = ['d-M-Y']  # format for displaying date
DATETIME_FORMAT = ['d-M-Y H:i:s']
```

I have been through documentation but couldn't figure out if there is a way to configure Viewflow for correct date format.

**Comment by kmmbvnr** [comment]
I don't think that is viewflow related. Serena any Django form would produce the error with that settings set.

**Comment by breneser** [comment]
I think you are right, so I tested with exact same datetime field in https://github.com/viewflow/viewflow/blob/master/viewflow/forms.py

I used 

```
    tada = forms.DateTimeField(widget=forms.HiddenInput,
                               initial=datetime.datetime.now())

```
and I got the same format.

But when I took out the widget when input is rendered I got the correct value

```
<input type="text" name="tada" value="18-Mar-2020 12:27:39" required="" id="id_tada">
```

So I guess problem occurs when hidden input is used. I suppose I could use a custom template to fix this. But wouldn't that be the case for everyone using Viewflow?

**Comment by breneser** [comment]
> I don't think that is viewflow related. Seems any Django form would produce the error with that settings set.
> 
> https://docs.djangoproject.com/en/3.0/ref/settings/#date-format
> 
> Date_format should be a string that matched one of input formats

Thanks for your answer. Changing DATE_FORMAT to string doesn't really change anything, it still puts in incorrect date format.

I can see it is a Django related mechanism. It seems hidden input widget doesn't respect date format settings. I will investigate further to be sure that is the case.

**Comment by breneser** [comment]
Further investigation shows I am having this problem because I am using Jinja2 templates and it seems dates in the templates are not formatted by `DATE_FORMAT` or `DATETIME_FORMAT` unlike django templates.

This specific problem I had doesn't exactly sound like a Viewflow problem. But I think there actually is an issue:

Firstly widget template for hidden input uses `value="{{ widget.value|stringformat:'s' }}"` which converts the object to string respecting `DATE_FORMAT` setting. And this is a huge problem in two ways:

1-) `DATE_FORMAT` and `DATETIME_FORMAT` are used for displaying dates, and if it doesn't match with one of `DATE_INPUT_FORMATS` -  which you said it should - it would cause an issue. My understanding from https://docs.djangoproject.com/en/3.0/ref/settings/#date-format is that `DATE_FORMAT` is for display purposes and it doesn't have to match with one of `DATE_INPUT_FORMATS`.

2-) Django uses OS locale over `DATETIME_FORMAT` if `USE_L10N` is enabled  in settings. Which would also break management_form if locale format was different.

So basically `DATE_FORMAT` and  `DATETIME_FORMAT` are used for display purposes, and hidden input doesn't really take date inputs into account. Not sure if it should; if it should then this can probably be reported to Django. 

If not though I think Viewflow needs to use `DATE_INPUT_FORMATS` and with a custom widget template formatting the date by `DATE_INPUT_FORMATS`  because the date in activation form is an input.

Would you accept a PR for this? 

@kmmbvnr  Please correct me if I get any of these wrong.
